### ISSUE TITLE
[Bug]: 

### ISSUE THREAD

**Issue by njgarnett** [jasp-stats/jasp-issues#1783]
### JASP Version

0.16.3

### What analysis are you seeing the problem on?

All analyses

### What OS are you seeing the problem on?

All

### Bug Description

JASP does not work with screen readers. This is really problematic for visually impaired users and may mean we are unable to use JASP in teaching.

### Expected Behaviour

JASP should be compatible with screen readers

### Steps to Reproduce

1. Try and use a screen reader with JASP
2. 
3. 
...

### Final Checklist

- [X] I have included a screenshot showcasing the issue, if any.
- [X] I have included a JASP file that causes the crash/bug, if any.
- [X] I have accurately described the bug, and steps to reproduce it.

**Comment by EJWagenmakers** [comment]
There was also a statement we can adopt which is useful for universities who want to install and use JASP. I believe our Jamovi friends have an example ready. @JTPetter could you look at this statement? I'll see whether I can dig that up from the requests...

**Comment by SwanseaUniversityTranscriptionCentre** [comment]
Is there any update on the progress of this? From September, we will be supporting a visually impaired student who uses a screen reader and will be using JASP.

**Comment by tomtomme** [comment]
@SwanseaUniversityTranscriptionCentre 
Sorry, no timeline yet for this feature.

**Comment by tomtomme** [comment]
@njgarnett @SwanseaUniversityTranscriptionCentre 
Would https://cran.r-project.org/web/packages/BrailleR/index.html be a good R-Package to implement this? Or do you have any experience with alternatives?

**Comment by TarandeepKang** [comment]
Hi All,

I don't know how to build software at all, but think dressing is built using QT and if I look at aspects of accessibility I reach a page like this which I don't understand. But someone here might? https://doc.qt.io/qt-5/accessible.html.

PS, I am a person with mobility disabilities so I have experience of using assistive software, but not designing software for use with it. I have a horrible feeling that this might be quite a big job, I hope to be proven wrong though!

**Comment by JohnnyDoorn** [comment]
Hi @SwanseaUniversityTranscriptionCentre,

Did you manage to work out some solution for your visually impaired student? I am currently trying to assist a student myself and was wondering if you found any solution here. 

@boutinb  @EJWagenmakers  @RensDofferhoff I do think this is an extremely important issue that would help a wider audience to adopt JASP, and I feel uncomfortable telling students to use JASP if we cannot accommodate visually impaired users. 

Cheers,
Johnny

**Comment by SwanseaUniversityTranscriptionCentre** [comment]
Hi Jordan,

Thank you for keeping us in mind.

We haven't come up with a solution other than a sighted lab assistant for our visual impaired students, which comes out of the DSA (Disabled Students Allowance).

Please do let us know of any progress with screen reader accessibility (or any other suggestions!).

Kind regards,

Angharad

**Comment by tomtomme** [comment]
@mosesrivera100 
Thx for your initiative. Do you have some programming knowledge in R and C++/Qt to check the code of Codex?
It needs to be readable and maintainable by others of the team in the long run.

**Comment by shun2wang** [comment]
thanks all of you. I will have a look.

**Comment by JorisGoosen** [comment]
Hey @mosesrivera100 
nice you wanna help out!

Currently the big blocker is that I can't find a way to access/enable the screenreader inside https://doc.qt.io/qt-6/qtwebenginequick.html (which is what we use to show the results).
The problem being that while it *is chromium basically* it should support screen readers somehow.
But I can't find the API... If you can find a way to turn on the screen reader in there then its solved!

Implementing a screen reader for the rest of the application is really easy actually. But pretty much pointless without the results, which is why I didnt even seem to have pushed my earlier test-branch.

Jamovi has it "implemented" by being run *completely* in a browser instead of having a browser inside of it.
But then via electron and not webengine, which while both being  chromium (I think?) are quite a different beast. So I wouldnt hope on too much inspiration on that count.

**Comment by TarandeepKang** [comment]
Hi @JorisGoosen I didn't find anything of any use on the documentation page for the webengine but I did manage to find these https://doc.qt.io/qt-6/accessible.html and  https://doc.qt.io/qt-6/qml-qtquick-accessible.html Which I guess you might already seen? Does look like you might need to add an accessibility  tag to add to each of your interface elements, that is if this thing works in the way I think it does!? It's also going to be a huge piece of work if so?

Also I am evidently not Moses! :-)

**Comment by TarandeepKang** [comment]
Hi @JorisGoosen My apologies you were way ahead of me. But this time I might have actually found something useful. 


I found this buried somewhere in the release notes about enabling accessibility -related features
https://doc.qt.io/qt-6/qtwebengine-platform-notes.html


Accessibility and Performance
Qt WebEngine enables accessibility support for web pages when the following conditions are met:

Qt Core is configured and built with accessibility support enabled.
The QPA plugin is notified by the operating system that accessibility should be activated. This happens for example when using a screen reader application on Windows or VoiceOver on macOS.

**Comment by TarandeepKang** [comment]
You'll find a way one way or other! :-)
